### PR TITLE
✨ Add Android plugin

### PR DESCRIPTION
**Pull request by davidbilik**
Adds android plugin with sample app. jCenter upload will be subject of another PR.

### CODE REVIEW

File: android/library/src/main/java/com/ackee/flipper/grpc/FlipperGrpcInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+) : ClientInterceptor {
+
+    override fun <ReqT : Any?, RespT : Any?> interceptCall(
+        method: MethodDescriptor<ReqT, RespT>,
+        callOptions: CallOptions,
+        next: Channel
+    ): ClientCall<ReqT, RespT> {
+        val id = UUID.randomUUID().toString()
+
+        return object : ForwardingClientCall.SimpleForwardingClientCall<ReqT, RespT>(
```
Comment (mottl-jan) [inline android/library/src/main/java/com/ackee/flipper/grpc/FlipperGrpcInterceptor.kt:21]: what about separate class definiton inside the file?

File: android/library/src/main/java/com/ackee/flipper/grpc/FlipperGrpcInterceptor.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                method,
+                callOptions
+            )
+        ) {
+
+            private var headers: Metadata? = null
+
+            override fun start(responseListener: ClientCall.Listener<RespT>, headers: Metadata) {
+                this.headers = headers
+                class LoggingClientCallListener :
```
Comment (mottl-jan) [inline android/library/src/main/java/com/ackee/flipper/grpc/FlipperGrpcInterceptor.kt:32]: I did not even know you can do this like in Java :D Wouldn't it be better to move it outside of the method body? :D

File: android/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    implementation "org.jetbrains.kotlinx:kotlinx-coroutines-android:1.3.7"
+
+    implementation 'androidx.core:core-ktx:1.3.1'
+    implementation 'androidx.appcompat:appcompat:1.2.0'
+    implementation 'com.google.android.material:material:1.2.0'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+
+    implementation 'io.grpc:grpc-okhttp:1.31.0'
+    implementation 'io.grpc:grpc-protobuf-lite:1.31.0'
+    implementation 'io.grpc:grpc-stub:1.31.0'
```
[the remark this answers is not in the document]
Comment (davidbilik) [reply android/app/build.gradle]: 2a56df9906d9b0a01c9b5850ab0da7bc05d2db57 refactored whole dependencies.

File: android/app/src/main/java/com/ackee/flipper/grpc/sample/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        val btnMakeRequest = findViewById<Button>(R.id.btn_make_request)
+        btnMakeRequest.setOnClickListener {
+            val channel = OkHttpChannelBuilder
+                .forAddress(/* YOUR HOST */"10.0.2.2", 50051 /*YOUR PORT*/)
+                .usePlaintext()
+                .intercept(App.flipperGrpcPlugin.getInterceptor())
+                .build()
+            val greeterStub = GreeterGrpc.newBlockingStub(channel)
+
+            GlobalScope.launch {
```
[the remark this answers is not in the document]
Comment (davidbilik) [reply android/app/src/main/java/com/ackee/flipper/grpc/sample/MainActivity.kt]: 8c3ffcde88ff847c58c6b3392a4ea9295cbbdd88

File: android/library/src/main/java/com/ackee/flipper/grpc/FlipperGrpcInterceptor.kt (hunk 1 above)
Comment (davidbilik) [reply android/library/src/main/java/com/ackee/flipper/grpc/FlipperGrpcInterceptor.kt]: c10d37fc064ea28596684833d493db71120b54b3 separated but im not sure if its more readable

File: android/library/src/main/java/com/ackee/flipper/grpc/FlipperGrpcInterceptor.kt (hunk 2 above)
Comment (davidbilik) [reply android/library/src/main/java/com/ackee/flipper/grpc/FlipperGrpcInterceptor.kt]: c10d37fc064ea28596684833d493db71120b54b3

**Pull request merged**